changes/views.py

```diff
@@ -218,7 +218,15 @@ class ReceiveAdminOptout(generics.CreateAPIView):
 
     def post(self, request, *args, **kwargs):
 
-        source = Source.objects.get(user=self.request.user)
+        sources = Source.objects.filter(user=self.request.user)
+        if sources:
+            source = sources[0]
+        else:
+            source = Source.objects.create(**{
+                "name": self.request.user.get_full_name(),
+                "user": self.request.user,
+                "authority": "advisor"
+            })
```

Comment:
We can probably use `get_or_create` rather than doing this ourselves: https://docs.djangoproject.com/en/1.11/ref/models/querysets/#get-or-create